src/simulation.jl

```diff
@@ -10,11 +10,11 @@ function simulate_ogata(
     t = tmin
     while t < tmax
         B, L = ground_intensity_bound(pp, t + eps(t), h)
-        τ = B > 0 ? rand(rng, Exponential(inv(B))) : Inf
+        τ = B > 0 ? rand(rng, Exponential(inv(B))) : typemax(inv(B))
         if τ > L
             t = t + L
         elseif τ <= L
-            U = rand(rng, T)
+            U = rand(rng)
```

Comment:
```suggestion
            U = rand(rng, T)
```
This directly samples a variable with the right type `T` instead of `Float64`

Reply:
I am not sure that type T is relevant here. If I am correct, T is the type of the time variables. However, U is more an "intensity variable" (in particular it is compared with intensity_bound which has no reason to be of type T).

Reply:
Essentially we have to decide which number type we want to get for the times when simulating. It can either be
- the type of the temporal bounds tmin and tmax (that is the current approach, albeit imperfectly implemented)
- the type defined implicitly by the point process (for example the type of inv(lambda))
- the promotion of the two to the most generic type

It is an arbitrary design choice, we just need to pick one option. The reason I like the current behavior is because you can easily know the type ahead of time, even when you never interact with the point process.

Reply:
I understand what you mean now, and you're right, `U` is unitless here